### DISCUSSION TITLE
Display error when running plugin on unsupported version of Grafana

### DISCUSSION THREAD

**Discussion by marcusolsson** [grafana/grafana#38185] [Dashboards, Panels & Visualizations]
**What would you like to be added**:

A user-friendly message telling the user that the plugin they're trying to use isn't supported for their Grafana version.

For example, if the user tries to run a panel plugin with `grafanaDependency: ">=7.3.0"` in the `plugin.json` on Grafana 7.2, they should receive an error along the lines of "This plugin requires a more recent version of Grafana".

**Why is this needed**:

If a plugin in the plugin directory depends on features that are only available in more recent versions of Grafana, it will throw cryptic errors/exceptions that can be hard to troubleshoot for most users.

**Comment by ryantxu** [comment]
Related issue  -- the cli should check that it only installs valid versions and give warnings that a new version exists, but not supported.

The other thing to note is that the server in plugin.json is valid when the plugin is published -- we need something else to check the maximum version!

**Comment by marcusolsson** [comment]
I believe the grafana-cli already checks whether the version is valid for the current Grafana version. Not everyone installs plugins via the CLI however, and it shouldn't be a big fix to check the semver of a plugin inside Grafana as well.

We validate every plugin that gets submitted to be published against the plugin.json JSON schema. JSON Schema doesn't have support for semver (we check against a regex right now), but we can add it to the [plugin-validator](https://github.com/grafana/plugin-validator) as an extra check.

**Comment by ryantxu** [comment]
Excellent -- if the cli is doing that then we should update the server so it sends all versions, not just the ones that match -- we added that hack on the server side to avoid updating all the 7+ plugins on 6x grafana!

**Comment by pkolyvas** [comment]
We're in the process of moving Grafana feature requests to [GitHub Discussions](https://community.grafana.com/t/were-moving-grafana-feature-requests-to-github-discussions/50341/3). This issue is being moved to [Discussions](https://github.com/grafana/grafana/discussions) or has been moved from our issues list, depending on where you're seeing this message. 

 We hope to use the discussion thread to continue building use cases, and additional evidence based on engagement/voting, in order to help us prioritize this feature request later on down the road. 

As with all things Grafana, we value your input and participation in this project and appreciate your patience during this transitional period. Looking forward to the discussion!

**Comment by gelicia** [comment]
Hello, as you may have heard, we are [transitioning away from using discussions to discuss feature requests](https://github.com/grafana/grafana/discussions/73424). Due to the age and number of responses to this discussion, we are deciding to close it. If this is something you would like to see in Grafana, feel free to open an [issue](https://github.com/grafana/grafana/issues) so the discussion can continue. Thank you!
